### PR TITLE
10090 viewer seats tests bugs

### LINKED ISSUE #10090: Viewer seats tests - bugs
hey guys,

- [x] Wrong redirection for datasets (username.carto.lan...). Not an actual issue. There's a double redirection that is valid (although @javitonino is on it at another PR and we might get rid of it). -> #10102

- [x] When a viewer account clicks on a shared dataset and opens it, he sees the "Create Map" button, which leads to an error when clicking on it. We may want to take that out.
screen shot 2016-10-07 at 15 26 25

- [x] Still inside the dataset view, if you click on "preview", you see again the "create map" button
screen shot 2016-10-07 at 15 27 00

- [x] On another note, the admin shared a few maps with me, and none of them appear in my dashboard
screen shot 2016-10-07 at 15 38 26

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #10090.

@xavijam CR this, please.

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-actions-view.js [hunk 1]
```diff
@@ -30,7 +34,8 @@ module.exports = CoreView.extend({
     this.clearSubViews();
     this.$el.html(
       template({
-        hasGeometry: this._queryGeometryModel.hasValue()
+        hasGeometry: this._queryGeometryModel.hasValue(),
+        createMapEnabled: typeof this['_mapAction'] != 'undefined'
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-actions-view.js]: I feel `!!this._mapAction` could do the trick here.

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-actions-view.js]: And I'd use `canCreateMap` as attribute.

**Comment by juanignaciosl** [comment]
@xavijam last CR after semistandard checks and fixes for tests to work, please.

**Comment by xavijam** [comment]
One question, can a viewer apply a SQL?

**Comment by juanignaciosl** [comment]
@xavijam yes, they can (confirmed with @saleiva, cc @javisantana ). Read-only operations (as they can only get read permissions).

This is ready to merge after your +1 if nothing else comes to your mind, @xavijam .

**Comment by xavijam** [comment]
:+1:

**Comment by xavijam** [comment]
If Travis tests are ok, you are ready to go.

File: lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-actions-view.js (hunk 1 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-actions-view.js]: Oh, not for me in the Conversation tab, but it's good to know that you do see it :-)

**Pull request merged**